src/main/java/com/minelittlepony/model/player/ModelAlicorn.java

```diff
@@ -0,0 +1,164 @@
+package com.minelittlepony.model.player;
+
+import com.minelittlepony.model.components.UnicornHorn;
+import com.minelittlepony.render.PonyRenderer;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.EnumHandSide;
+import net.minecraft.util.math.MathHelper;
+
+import com.minelittlepony.model.capabilities.IModelUnicorn;
+
+import static com.minelittlepony.model.PonyModelConstants.*;
+
+/**
+ * Used for both unicorns and alicorns since there's no logical way to keep them distinct and not duplicate stuff.
```

Comment:
It's OK to duplicate stuff. You have interfaces, so that's fine.

Reply:
Problem is when the duplication involves copying _the logic_, and about a couple hundred lines of code with it. Also duplication means which one do you write twice? And who's toes do you have to bludgeon when someone invariably updates one and forgets to update the other?

Reply:
Meh, on second though, I may revisit this at a later date. With how I've implemented pegasus wings it shouldn't be too bad to apply.